includes/wpmenucart-settings.php

```diff
@@ -568,299 +573,272 @@ public function display_pro_ad() {
 	 * @return array menu slug => menu name
 	 */
 	public function get_menu_array() {
-		$menus     = get_terms( 'nav_menu', array( 'hide_empty' => false ) );
+		$menus     = get_terms( array(
+			'taxonomy'   => 'nav_menu',
+			'hide_empty' => false,
+		) );
 		$menu_list = array();
 
 		foreach ( $menus as $menu ) {
 			$menu_list[$menu->slug] = $menu->name;
 		}
-		
+
 		if ( ! empty( $menu_list ) ) {
 			return $menu_list;
 		}
 	}
-	
+
 	/**
 	 * Get array of active shop plugins
 	 * 
 	 * @return array plugin slug => plugin name
 	 */
 	public function get_shop_plugins() {
 		$active_shop_plugins = WpMenuCart::get_active_shops();
-		
+
 		//switch keys & values, then strip plugin path to folder
 		foreach ( $active_shop_plugins as $key => $value ) {
 			$filtered_active_shop_plugins[ dirname($value) ] = $key;
 		}
 
 		$active_shop_plugins = isset( $filtered_active_shop_plugins ) ? $filtered_active_shop_plugins : '';
-				
+
 		return $active_shop_plugins;
 	}
 
 	/**
 	 * Text field callback.
 	 *
-	 * @param  array $args Field arguments.
+	 * @param array $args Field arguments.
 	 *
-	 * @return string	  Text field.
+	 * @return void      Text field.
 	 */
-	public function text_element_callback( $args ) {
-		$menu = $args['menu'];
-		$id   = $args['id'];
-		$size = isset( $args['size'] ) ? $args['size'] : '25';
-		$pro  = isset( $args['pro'] ) ? $args['pro'] : false;
-	
-		$options = get_option( $menu );
-	
-		if ( isset( $options[$id] ) ) {
-			$current = $options[$id];
-		} else {
-			$current = isset( $args['default'] ) ? $args['default'] : '';
-		}
+	public function text_element_callback( array $args ): void {
+		$menu     = $args['menu'];
+		$id       = $args['id'];
+		$size     = $args['size'] ?? '25';
+		$pro      = $args['pro'] ?? false;
+		$options  = get_option( $menu );
+		$current  = $options[ $id ] ?? $args['default'] ?? '';
+		$disabled = ( isset( $args['disabled'] ) ) ? ' disabled=""' : '';
```

Comment:
I saw that `wp_kses` strips out the `disabled` attribute when a value is not provided, so I have to set a value for it. Since the whole key value, not only the value, was sent to the `sprintf`, I removed the escaping.

After some tests, I understood that the `disabled` attribute was stiped out in `<input disabled/>` because the parser incorrectly treated the `disabled/` as part of an attribute or misinterpreted the tag closure. So, adding a white space before `/>` fixed the issue, like this: `<input disabled />` .

I'll revert these changes.